src/main/java/org/opensearch/security/ssl/DefaultSecurityKeyStore.java

```diff
@@ -991,19 +991,34 @@ private SslContext buildSSLServerContext(
         final SslProvider sslProvider,
         final ClientAuth authMode
     ) throws SSLException {
+        final SecurityManager sm = System.getSecurityManager();
 
-        final SslContextBuilder _sslContextBuilder = configureSSLServerContextBuilder(
-            SslContextBuilder.forServer(_cert, _key, pwd),
-            sslProvider,
-            ciphers,
-            authMode
-        );
-
-        if (_trustedCerts != null) {
-            _sslContextBuilder.trustManager(_trustedCerts);
+        if (sm != null) {
+            sm.checkPermission(new SpecialPermission());
         }
 
-        return buildSSLContext0(_sslContextBuilder);
+        try {
+            final SslContextBuilder _sslContextBuilder = AccessController.doPrivileged(new PrivilegedExceptionAction<SslContextBuilder>() {
+                @Override
+                public SslContextBuilder run() throws Exception {
+                    return SslContextBuilder.forServer(_cert, _key, pwd)
+                        .ciphers(ciphers)
+                        .applicationProtocolConfig(ApplicationProtocolConfig.DISABLED)
+                        .clientAuth(Objects.requireNonNull(authMode)) // https://github.com/netty/netty/issues/4722
+                        .sessionCacheSize(0)
+                        .sessionTimeout(0)
+                        .sslProvider(sslProvider);
+                }
+            });
+
+            if (_trustedCerts != null) {
+                _sslContextBuilder.trustManager(_trustedCerts);
+            }
+
+            return buildSSLContext0(_sslContextBuilder);
+        } catch (final PrivilegedActionException e) {
+            throw (SSLException) e.getCause();
```

Comment:
This cast could fail (which would be very confusing), instead check if it is an instance of SSLException and then throw, if not rethrow as a runtime exception

Reply:
Added the check in. If it fails the check then it throws a RuntimeException